### PR TITLE
Bump GHC 927 -> 962

### PR DESCRIPTION
**Pull request by zeme-wana**

### CODE REVIEW

**Comment by zeme-wana** [comment]
@jorisdral this can be completed now. You can enter a `nix develop` shell with ghc962

File: quickcheck-dynamic-iosim/quickcheck-dynamic-iosim.cabal [hunk 1]
```diff
@@ -56,7 +56,7 @@ library
   hs-source-dirs:  src
   exposed-modules: Test.QuickCheck.StateModel.IOSim
   build-depends:
-    , base                >=4.7 && <5
```
Comment (jorisdral) [inline quickcheck-dynamic-iosim/quickcheck-dynamic-iosim.cabal:59]: I'm not sure we would should remove these bounds

**Review by jorisdral — COMMENTED** [review]
I don't have access to this repository, so I've include a patch instead, which can be found below. The compilation error is fixed by https://github.com/input-output-hk/quickcheck-dynamic/pull/48/files, but it isn't merged yet, so the patch includes the change from #48. Also,  I've suggested adding `ghc-8.10.7` and `ghc-9.2.8` to the list of supported compilers, such that Hydra uses each of them.

<details>
<summary>fix.patch</summary>

```text
diff --git a/nix/haskell.nix b/nix/haskell.nix
index 4bc733a..eec0c8c 100644
--- a/nix/haskell.nix
+++ b/nix/haskell.nix
@@ -2,5 +2,6 @@
 # https://www.github.com/input-output-hk/iogx#32-nixhaskellnix
 
 {
-  supportedCompilers = ["ghc962"];
+  supportedCompilers = ["ghc8107" "ghc928" "ghc962"];
+  defaultHaskellCompiler = "ghc928";
 }
\ No newline at end of file
diff --git a/quickcheck-dynamic/src/Test/QuickCheck/StateModel.hs b/quickcheck-dynamic/src/Test/QuickCheck/StateModel.hs
index 9836084..3932923 100644
--- a/quickcheck-dynamic/src/Test/QuickCheck/StateModel.hs
+++ b/quickcheck-dynamic/src/Test/QuickCheck/StateModel.hs
@@ -379,6 +379,7 @@ instance StateModel state => Arbitrary (Actions state) where
   shrink (Actions_ rs as) =
     map (Actions_ rs) (shrinkSmart (map (prune . map fst) . shrinkList shrinker . withStates) as)
     where
+      shrinker :: (Step state, Annotated state) -> [(Step state, Annotated state)]
       shrinker (v := act, s) = [(unsafeCoerceVar v := act', s) | Some act'@ActionWithPolarity{} <- computeShrinkAction s act]
 
 -- Running state models
```

</details>

**Comment by zeme-wana** [comment]
I applied all requested changes: follow progress here https://ci.iog.io/eval/9992

**Pull request merged**